students/uss_tomasz/lesson_12_debugging/12_3_valdated_user_base/validate_user_base.py

```diff
@@ -0,0 +1,120 @@
+    [\s-]?
+    \d{3}
+    )
+    )''', re.VERBOSE)
+ZIP_CODE_REGEX = re.compile(r'\d\d-\d\d\d')
+REGEX_DATABASE = {'email': EMAIL_REGEX, 'password': PASSWORD_REGEX, 'telephone': TELEPHONE_REGEX,
+                  'zip_code': ZIP_CODE_REGEX}
+
+
+def parse_args(args):
+    parser = argparse.ArgumentParser('Collect database of mail, pass, tel and zip inside csv.')
+    parser.add_argument('-e', action='store', dest='email', help='Email of user. This is unique filed for database.')
+    parser.add_argument('-p', action='store', dest='password', help='Password of user.')
+    parser.add_argument('-t', action='store', dest='telephone', default='000-000-000', help='Telephone of user.')
+    parser.add_argument('-z', action='store', dest='zip_code', default='00-000', help='Zip code of user.')
+    received_args = parser.parse_args(args)
+    if received_args.email is None or received_args.password is None:
+        return {}
+    else:
+        results = {'email': received_args.email, 'password': received_args.password,
+                   'telephone': received_args.telephone, 'zip_code': received_args.zip_code}
+        return results
+
+
+def is_record_correct(record_type, record):
+    return re.match(REGEX_DATABASE[record_type], record) is not None
+
+
+def validate_registration(registration):
+    for position in registration:
+        if not is_record_correct(position, registration[position]):
+            return False
+    return True
+
+
+def csv_existence_checker(csv_file_name):
+    if not os.path.isfile(csv_file_name):
+        with open(csv_file_name, 'w', newline='') as csv_file:
+            writer = csv.writer(csv_file)
+            writer.writerow(['email', 'password', 'telephone', 'zip-code'])  # columns names
+
+
+def get_csv_content(csv_filename):
+    csv_content = []
+    with open(csv_filename) as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=',')
+        for row in csv_reader:
+            csv_content.append(row)
+    return csv_content
+
+
+def is_registration_inside_csv(csv_file_name, registration):
+    csv_content = get_csv_content(csv_file_name)
+    line_count = 0
+    for row in csv_content:
+        if line_count == 0:
+            pass  # columns names
+        else:
+            if row[0] == registration['email']:
+                logging.log(logging.WARNING, "This registration is already inside file.")
+                return True
+        line_count += 1
+    logging.log(logging.INFO, "This registration is new inside CSV file.")
+    return False
+
+
+def csv_entry(csv_file_name, registration):
+    with open(csv_file_name, 'a', newline='') as csv_file:
+        writer = csv.writer(csv_file)
+        writer.writerow([registration['email'], registration['password'], registration['telephone'],
+                         registration['zip_code']])
+
+
+def execute_script(args, csv_file_name):
+    csv_existence_checker(csv_file_name)
+    user_input = parse_args(args)
+    if len(user_input) == 0:
```

Comment:
you returned `{}` and then check if `len({}) == 0`

There is no need to elevate the complexity this way. Consider more straightforward solutions:
1. Put required in argparse (suggested solution)
2. return None and check `if user_input:`